PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,542 @@
+        else:
+            result = self.delete(val)
+            if result != None:
+                msg = "{} Deleted!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+        self.setMessage(msg)
+        self.clearArgument()
+        
+    def clickDeleteFirst(self):
+        val = self.validateArgument()
+        result = self.deleteFirst()
+        if not self.first: 
+            msg = "ERROR: Linked list is empty"
+        else:
+            msg = "first node deleted"
+        self.setMessage(msg)
+        self.clearArgument()
+        
+        
+    def clickNewLinkedList(self):
+        self.canvas.delete('all')
+        self.newLinkedList()
+    
+    def clickGetFirst(self):
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
+    
+
+    def makeButtons(self):
+        vcmd = (self.window.register(validate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        searchButton = self.addOperation(
+            "Search", lambda: self.clickFind(), numArguments=1,
+            validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1,
+            validationCmd=vcmd)
+        deleteButton = self.addOperation(
+            "Delete", lambda: self.clickDelete(), numArguments=1,
+            validationCmd=vcmd)
+        self.addAnimationButtons()
+        deleteFirstButton = self.addOperation("Delete First", lambda: self.clickDeleteFirst(), 
+                                              numArguments = 0, validationCmd = vcmd, maxRows = 3)
+        newLinkedListButton = self.addOperation(
+            "New", lambda: self.clickNewLinkedList(), 
+            numArguments = 0, validationCmd =vcmd, maxRows = 3)
+        getFirstButton = self.addOperation(
+            "Get First", lambda: self.clickGetFirst(), numArguments = 0,
+            validationCmd=vcmd, maxRows = 3)
+
+    
+        return [searchButton, insertButton, deleteButton, deleteFirstButton, newLinkedListButton, getFirstButton]        
+
+    
+    # validate text entry
+    def validateArgument(self):
+        text = self.getArgument()
+        if text:
+            if text.isdigit():
+                val = int(text)
+                if val< 100000000: return val
+            elif len(text)<9: return text
+        else:
+            return -1   
+            
+#allow letters or numbers to be typed in                  
+def validate(action, index, value_if_allowed,
+            prior_value, text, validation_type, trigger_type, widget_name):
+    if not(value_if_allowed.isdigit()):
+        for i in value_if_allowed:
+            if not 65<=ord(i)<=122 and i not in "0123456789": return False
+    return True
```

Comment:
The `validate()` function is too strict.  There's no harm in allowing whitespace, punctuation, even emoji's to be entered.  The only restriction should be on the length of the text.  Use some reasonable constant, like 8, and pass that as the `maxArgWidth` in the call to construct the `VisualizationApp`.  That will make the text entry widget wider to show all the characters.